### PR TITLE
Patch for the After/before load PR  validation issue

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

_Select the type of your PR_

- Bugfix

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 1]
```diff
@@ -648,7 +648,7 @@ public MapAndStatus(T loadMap, K status) {
         return new MapAndStatus<>(rebalanceMap, proposalJson.getJsonObject(CruiseControlRebalanceKeys.SUMMARY.getKey()).getMap());
     }
 
-    private MapAndStatus<ConfigMap, KafkaRebalanceStatus> buildRebalanceStatus(KafkaRebalance kafkaRebalance, String sessionID, KafkaRebalanceState cruiseControlState, JsonObject proposalJson) {
+    private MapAndStatus<ConfigMap, KafkaRebalanceStatus> buildRebalanceStatus(KafkaRebalance kafkaRebalance, String sessionID, KafkaRebalanceState cruiseControlState, JsonObject proposalJson, Set<Condition> validation) {
```
Comment (sknot-rh) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:651]: You added the validation argument here but it does not seem to be used anywhere.

Comment (ShubhamRwt) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:651]: I have made the fixes now

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java [hunk 2]
```diff
@@ -175,7 +173,6 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
                         System.out.println(result.getStatus().getConditions());
```
Comment (tomncooper) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java:173]: You need to remove this print statement as well.

**Review by sknot-rh — COMMENTED** [review]
I am still missing the NPE fix when one of the goals is missing.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 3]
```diff
@@ -701,7 +704,7 @@ public MapAndStatus(T loadMap, K status) {
             return onNew(reconciliation, host, apiClient, kafkaRebalance, rebalanceOptionsBuilder);
         } else {
             // Stay in the current NotReady state, returning null as next state
-            return Future.succeededFuture();
+            return Future.succeededFuture(new MapAndStatus<>(null, null));
```
Comment (ShubhamRwt) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:707]: @sknot-rh Hello, this line seems to resolve the NPE.

Comment (sknot-rh) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:707]: Aha. Thank you.

**Review by sknot-rh — APPROVED** [review]

**Review by tomncooper — APPROVED** [review]
LGTM

**Review by scholzj — APPROVED** [review]

**Pull request merged**